### ISSUE TITLE
Site Editor redirects to New Post Editor instead

### ISSUE THREAD

**Issue by cuemarie** [Automattic/wp-calypso#63262]
### Quick summary

User reported that when they go to **Appearance > Editor**, they're being redirected to `/wp-admin/post-new.php?post_type=post&calypsoify=1`

I was able to reproduce on my end as well, but only on their site at the moment.

### Steps to reproduce

1. Open user's site via SU
2. Select Appearance > Editor (in same window)

### What you expected to happen

Site Editor opens

### What actually happened

Browser redirects to new post editor:

https://user-images.githubusercontent.com/27249804/168708874-3524d74d-d0e9-40cf-afe4-b5581dc0dcc7.mp4

### Context

35322277-hc

### Simple, Atomic or both?

Atomic

### Theme-specific issue?

Blank Canvas

### Browser, operating system and other notes

I was able to reproduce in FireFox on Mac OS

### Reproducibility

Intermittent

### Severity

One

### Available workarounds?

Yes, easy to implement

### Workaround details

Opening the Site Editor in a new tab works as expected. If that doesn't work, open the Site Editor via the Edit button on the Themes Calypso page as in this screenshot, which redircts to the WP-Admin Site Editor page at `wp-admin/site-editor.php`: 
![Edit on the Themes page](https://user-images.githubusercontent.com/1519466/182807829-932d3eb3-1932-4108-8b2c-a2d5e3824647.png)

**Comment by cuemarie** [comment]
Got a little further with testing this in the chat - switching the primary address to their default domain seems to have resolved the issue. (User hasn't been able to confirm yet due to other issues, but thinking this was a domain thing). Closing this!

**Comment by cuemarie** [comment]
New report of the same issue here: 5217168-zd-woothemes

Looking back at the original chat I had for which I opened this report, I'm not sure that changing the primary domain was the solution after all - the chat kept going after I had it, and the user continued to run into browser issues for a bit.

Regarding this new issue, 5217168-zd-woothemes - I SSP'd into the site and was able to replicate the issue:

1. Click on Site Editor from Appearance Menu
2. In the address bar, I can see the site try to load `wordpress.com/log-in.php`, then redirect to `/wp-admin/post-new.php?post_type=post&calypsoify=1` - this matches the previous user's explanation as well

<img width="1037" alt="Markup 2022-05-16 at 18 49 30" src="https://user-images.githubusercontent.com/27249804/168712036-f29f9d7a-aa1f-44e3-b53d-a3355a599f6e.png">

<img width="1042" alt="Markup 2022-05-16 at 18 50 24" src="https://user-images.githubusercontent.com/27249804/168712051-388f8a0e-d93c-4eab-8f28-60e7d0ac6345.png">

<img width="1035" alt="Markup 2022-05-16 at 18 51 08" src="https://user-images.githubusercontent.com/27249804/168712062-0709c580-d51a-44d1-9917-3756e9f54ec5.png">

Following this test, I SSP'd into the OP user's site, but the issue is no longer occurring there. I did this using the same browser - Safari - which makes me think the issue is not a browser issue.

Reopening for further investigation.

**Comment by arunsathiya** [comment]
I am unable to see this behavior on an Atomic site with Blank Canvas theme. I do wonder if this is related to the to/from FSE theme changes that's described here:

- https://github.com/Automattic/wp-calypso/issues/62755

Let's leave this issue open for a bit more to track similar reports.

**Comment by kdevnel** [comment]
I have another example of this happening with Twenty Twenty Two. From Calypso the first time I visited it worked fine clicking Appearance > Editor beta.

From then, clicking there from Calypso always opens a new post. If I try from WP Admin, the Appearance > Editor beta button goes to Site Editor correctly.

Reference: p1653038963164269-slack-C03TY6J1A and 35557163-hc

**Comment by ClassicRKR27** [comment]
5216679-zd-woothemes

This site is using the `Attar`. I found that the above workarounds work correctly - either exiting the `New Post` and trying again brings you to the `Site Editor`, or the more effective method of right-clicking the `Editor` link and opening it in a new tab works as well.

**Comment by cuemarie** [comment]
📌 HOUSEKEEPING
- Labels ✅ 
- Priority ✅ 
- Replicable on Core - No

📌 SCRUBBING
- Tested on Simple ✅ 
- Tested on AT ✅ 
- Tested on Self-hosted ✅ 

📌 FINDINGS/SCREENSHOTS/VIDEO
- **So far, we've yet to find a way to replicate this issue. The only consistency I notice on the above reports is that they're all Atomic.**
- I've tried switching primary domains right before trying to access Site Editor, as well as setting a domain with the wrong name servers as my primary domain, but still haven't been able to replicate the issue.
- I also checked all 4 sites that reported this issue above, and the issue does not appear to be happening anymore.

📌 ACTIONS
- Closed as could not replicate, please reopen this issue if it reoccurs

If others run into this issue, feel free to reopen with any additional steps you can track to replicate. Check the BRC for any changes or updates that might trigger this issue.

In the meantime, with all the cases above, the issue has cleared up on it's own in a matter of time.

**Comment by that-mike-bal** [comment]
I ran into this issue with staging sites for themes that are being worked on by a vendor and @KirkwallDay helped me out. 

Caught a recording of it on one site: 
![Weird redirect from editor](https://user-images.githubusercontent.com/89872780/174163205-60bffa44-6f42-4535-a20d-9f76fff67e90.gif)

 James found some issues on his end before asking me to click a link to reconnect Jetpack which seemed to fix the issue. 

All sites were being built off of Zoologist and were working fine as of yesterday.

**Comment by KirkwallDay** [comment]
For further context, the issue here is that Jetpack debug was saying the URLs connected were the .wordpress.com URL's rather then the .wpcomstaging.com URL's. It's unclear why this was, and I had to reconnect Jetpack on the two affected sites @that-mike-bal reported having issues with. The reconnect flow was also different than normal, requiring me to reconnect once, and then hit "Reconnect Jetpack" in the Jetpack Settings screen and go through a reconnect flow again.

tl;dr: I think the sites that are experiencing the issues are connected to Jetpack with incorrect site URL's

**Comment by KirkwallDay** [comment]
Forgot to re-open this, in case this new info helps anyone find out why the sites had the incorrect site URL when going AT.

**Comment by goblinartificer** [comment]
I have a new instance of this, in 35911200-hc and related previous chats. User switched back and forth to/from FSE (one theme used was Argent). Disabling Gutenberg plugin fixed this for _me_ but I did have to reconnect Jetpack to get it to work for the user

**Comment by happychait** [comment]
Another report in 35882779-hc

**Comment by cuemarie** [comment]
This issue presented for myself and the next HE at the end of this chat: 36018994-hc

Could consistently replicate the issue on the user's site. I checked JP Debugger for sign of the wrong connected URLs per @KirkwallDay 's note above, but I anything there that didn't match my other test sites. 

That said, **disconnecting Jetpack and Reconnecting resolved the issue on my end** - letting user know in follow up here: 5360428-zd-woothemes

**Comment by rw-ye** [comment]
5362554-zd-woothemes

I tried disconnecting and reconnecting Jetpack and that appears to have fixed it.

Awaiting confirmation from the user.

**Comment by edequalsawesome** [comment]
Had the same issue in 36080116-hc (Chrome on macOS) -- disconnecting and reconnecting Jetpack worked for the customer

**Comment by dolgelukkig** [comment]
36005436-hc

Disconnecting and reconnecting Jetpack didn't solve the issue.

**Comment by deancroyal** [comment]
36119891-hc

Was able to reproduce on this user's site. Steps...

1. From Dashboard, click on Editor within Appearance.
2. Editor redirects to New Page Editor.
3. Click the back button in browser.
4. Right click on Site Editor and copy link.
5. Open link in new tab.
6. Editor opens as expected.

h/t @sophiegyo for the find

**Comment by cuemarie** [comment]
This issue happened for me in SU during this chat: 36165162-hc

Holding CMD+clicking Editor opened the Site Editor correctly in a new tab. User did not report the issue

**Comment by jordesign** [comment]
Seeing this happening in 36285160-hc

Switching theme away, and then back seemed to resolve it.
This took place on a Blockbase child theme

**Comment by thesacredpath** [comment]
Have another instance of this. User can reach the FSE editor just fine, but if I go to Appearance FSE it redirects me to the post editor. I was using support session and did not try to see if SSP'ing as the user would fix the issue, but I have my doubts it would.

User gave me direct link from browser address bar and I was able to get to FSE that way.

Simple site: Personal Plan - Arbutus theme
Chat: 35471744-hc
Follow-up: 5450195-zen

**Comment by inaikem** [comment]
36119882-hc

Noting that for this interaction, the user had both a potentially broken Blockbase child theme and had deleted the Gutenberg plugin from their site. Switching themes, opening the `Site editor` link in a new tab or other workarounds did. not work.

I advised them to reinstall Gutenberg but this was not done (following chat transfer) and the issue still persists.

An internal discussion thread (p1659583521466569-slack-C03TY6J1A) suggests this PR may be the cause of this (still checking):

https://github.com/Automattic/jetpack/pull/25281

**Comment by jeherve** [comment]
This may indeed be linked. The change became available on WordPress.com Simple sites 2 days ago, and on WoA yesterday.

Out of the 3 reports above, one seems to be on WordPress.com Simple, and 2 on WoA.

I've tried to reproduce like so:

#### On a WordPress.com Simple site using a WordPress.com Personal plan

1. Enable a Blockbase child theme (Zoologist)
2. Go to the Posts menu in Calypso
3. Toggle between the default and the classic view

<img width="315" alt="image" src="https://user-images.githubusercontent.com/426388/182770545-69233c9f-89a7-4414-aa21-683339e08332.png">

4. In the menu, go to Appearance > Editor (Beta)
<img width="548" alt="image" src="https://user-images.githubusercontent.com/426388/182770616-15d0937f-61de-4f49-93d9-633ab204d5c6.png">

I tried this in both default and classic view, and every time I am redirected to `mysite.wordpress.com/wp-admin/site-editor.php?postType=wp_template&postId=pub%2Fzoologist%2F%2Findex` as expected.

5. I tried the same site after toggling Gutenberg Edge on and off on that same site (Gutenberg Edge was just updated a few hours ago in #66094, I wanted to see if this would change anything), but the site editor remains accessible.

I do not experience any issues when accessing pages directly either:

- `https://wordpress.com/site-editor/mysite`
- `https://mysite.wordpress.com/wp-admin/site-editor.php`

#### On a WoA site (using a Business plan)

I followed the same steps as above on a WoA site (running Gutenberg 13.7.3), but could not reproduce either.

I do not experience any issues when accessing the wp-admin page directly either:

- `https://mysite.com/wp-admin/site-editor.php`

I did, however, get redirected when trying to access

- `https://wordpress.com/site-editor/mysite`

I was redirected to 
`https://mysite.com/wp-admin/post-new.php?post_type=post&calypsoify=1`

This may point us to an issue here:
https://github.com/Automattic/wp-calypso/blob/126e921bfc0640fb20fde16f52a42cf50273ab03/client/gutenberg/editor/calypsoify-iframe.tsx#L859-L862

***************

cc @fullofcaffeine, @noahtallen, and @creativecoder who worked on https://github.com/Automattic/jetpack/pull/25281 and who may have some more ideas.

- Related Calypso PR: #65997
- WordPress.com Simple: D84883-code

***************

Internal references:

- p9oQ9f-1i0-p2#comment-1941
- p9dueE-5tE-p2

**Comment by happychait** [comment]
- [ ] Another report to be followed up in 5452120-zen


Worked around and opened the Site Editor via the Edit button on the Themes page as in this screenshot: 
![Edit on the Themes page](https://user-images.githubusercontent.com/1519466/182806988-04bff8fc-7bc4-4b3e-af8d-d1a8df161596.png)

**Comment by brinbliss** [comment]
- [ ] Report here in 5452262-zd-woothemes

Followed up with the user regarding the current workaround within Appearance -> Themes -> "Edit" button.

**Comment by khristiansnyder** [comment]
Also ran into this today in 36330300-hc. 

Only workaround was to manually use the WP-Admin link: `wp-admin/site-editor.php`

**Comment by kelasante** [comment]
Just run into this in 36090045-hc. Changed the primary domain, but the issue persists. Only workaround was using `wp-admin/site-editor.php`

**Comment by fullofcaffeine** [comment]
@jeherve 

Quoting from your [comment above](https://github.com/Automattic/wp-calypso/issues/63262#issuecomment-1204787731):
> I did, however, get redirected when trying to access
>
>    https://wordpress.com/site-editor/mysite

That's a known issue (unfortunately), and is being tracked here: https://github.com/Automattic/wp-calypso/issues/66020.

**Comment by creativecoder** [comment]
I think I've found the cause of this. Working on a PR here: https://github.com/Automattic/wp-calypso/pull/66265

**Comment by creativecoder** [comment]
I think we've fixed this with https://github.com/Automattic/wp-calypso/pull/66265, which I'm in the process of deploying right now, but please reopen this issue if there are additional reports.

**Comment by Greatdane** [comment]
Hi everyone 👋 Just a quick note to let you all know @Automattic/dotcom-pa-group have finished following up with affected users reported in this thread.  

Thanks for all the reports!

**Comment by creativecoder** [comment]
Thank you @Greatdane !
